project/ScalafixBuild.scala

```diff
@@ -68,6 +68,16 @@ object ScalafixBuild extends AutoPlugin with GhpagesKeys {
       "-groups",
       "-implicits"
     )
+    lazy val xsource3 = Seq(
+      scalacOptions += "-Xsource:3",
+      Compile / unmanagedSourceDirectories :=
+        ((Compile / sourceDirectory).value * "*xsource3").get
```

Comment:
Not particular proud about that. For sure, it would be more maintainable to have a dedicated `inputSource3` project with a sources in the regular `scala`, but I haven't found a way to integrate that in the `unit` matrix without duplicating the entire testkit properties setup.